crates/networking/p2p/sync/storage_fetcher.rs

```diff
@@ -95,7 +118,9 @@ pub(crate) async fn storage_fetcher(
             .send(pending_storage.into_iter().map(|(hash, _)| hash).collect())
             .await?;
     }
-    Ok(())
+    // Signal large storage fetcher
+    large_storage_sender.send(vec![]).await?;
```

Comment:
Why do we need this signaling?

Reply:
The large storage fetcher (as well as other fetchers) act as a queue, waiting for incoming requests from their channel and fetching the associated storages. We send a empty batch to signal that we will not be sending any more requests, so the fetcher can finish up and return